storage/cloning.md

```diff
@@ -0,0 +1,166 @@
+# CSI volume cloning
+
+Volume cloning is the process of using an existing persistent volume claim (PVC) as a data source when creating a new one.  
+Some of the use-cases for cloning might be:
+- use as backup before upgrading application that uses volume by deploying new version of application using cloned version of the original volume
+- pre-populate volume with some data and use that data as base state by deploying new version of application using cloned version of the original volume
+
+
+Read more about [volume cloning](https://kubernetes.io/docs/concepts/storage/volume-pvc-datasource/#introduction) from Kubernetes documentation. 
+
+Cloning persistent volume requires following steps:
+- shut down application that uses persistent volume that is going to be cloned
+- create new persistent volume claim and define existing persistent volume claim as data source
+- when cloning is done, start application(s)
+
+
+## Prerequisites
+- `kubectl` installed
+- cluster config (kubeconfig)
+
+
+This document uses `mariadb` deployment from [README](README.md) as an example. To test this out, follow the steps from `README` and fill the database with test data (numbers).   
+Goal here is to use pre-populated (numbers from 1 to 4) data from `mariadb-pvc` as base numbers for new MariaDB instance and add some more numbers on top of those.
+
+## Shut down application
+Shut down the application that uses PVC that is going to be cloned. This will detach the storage device from the worker node and allow the back-end to clone the storage. 
+
+```shell
+$ kubectl -n default scale deployment mariadb --replicas=0
+deployment.apps/mariadb scaled
+```
+
+## Create persistent volume
+Save following content to file named `clone.yaml`
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: mariadb-pvc-clone
+spec:
+  storageClassName: upcloud-block-storage-maxiops
+  volumeMode: Filesystem
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
```

Comment:
Should we mention that this size can be different (bigger than or equal to) than the volume that's being cloned? Just for clarification

Reply:
Good point :+1:  I'll push changes.